File: longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string lcp = strs[0];
        for(int i = 1; i < strs.size(); i++)
        {
            int same = 0;
            for(int j = 0; j < strs[i].size(); j++) {
                if(strs[i][j] == lcp[j]) {
                    same++;
                }
                else {
                    break;
                }
            }
            lcp = lcp.substr(0,same);
        }
        return lcp;
    }
};

File: longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ""
        for i in range(min([len(s) for s in strs])):
            if (len(set([s[i] for s in strs])) == 1):
                ans += strs[0][i]
            else:
                break
        return ans
            